lib/validation.ts

```
@@ -0,0 +1,32 @@
+export async function validateFramePayload(payload: FrameRequest): Promise<FramePayloadValidated> {
+    switch (payload.clientProtocol) {
+        case 'farcaster':
+            return await validateFarcasterPayload(payload);
+        case 'lens':
+            const lensValidation = await validateLensFrame(payload as LensFrameRequest);
+            if (!lensValidation.isValid || !lensValidation.validatedPayload) {
+                throw new Error(lensValidation.error || 'Invalid Lens frame message');
+            }
+            return lensValidation.validatedPayload;
+        case 'xmtp':
+            const xmtpValidation = await validateFramesPost(payload);
```

Biome

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)